app/services/email_service/api/addresses.rb

```diff
@@ -0,0 +1,45 @@
+module EmailService::API
+  AddressStore = EmailService::Store::Address
+
+  class Addresses
+
+    def initialize(default_sender:)
+      @default_sender = default_sender
+    end
+
+    def get_sender(community_id:)
+      sender = Maybe(community_id).map {
+        AddressStore.get(community_id: community_id)
+      }.map { |address|
+        to_smtp_format(name: address[:name], email: address[:email])
+      }.or_else(@default_sender)
+
+      Result::Success.new(sender)
+    end
```

Comment:
My suggestion is to change this to be a hash with single key 'formatted'. This way it's ~the same format (with some keys missing) as what create and get_user_defined would use.